src/core/agent/records/ipexMessageRecord.ts

```diff
@@ -4,15 +4,17 @@ import { IpexMessage } from "../agent.types";
 interface IpexMessageProps {
   id: string;
   content: IpexMessage;
+  isUpdate?: boolean;
   createdAt?: Date;
   connectionId: string;
 }
 
 class IpexMessageRecord extends BaseRecord {
   content!: IpexMessage;
   connectionId!: string;
+  isUpdate?: boolean;
```

Comment:
Hmm, for extensibility, can we instead just store the the connection history type directly here?